src/cfml/system/services/ServerService.cfc

```diff
@@ -1037,14 +1037,14 @@ component accessors="true" singleton {
 
 		serverInfo.trayOptions.prepend( { 'label':'Open...', 'items': openItems, "image" : expandPath('/commandbox/system/config/server-icons/open.png' ) } );
 
-		// serverInfo.trayOptions.prepend( { 'label' : 'Restart Server', 'hotkey':'R', 'action' : 'restartserver', 'image': expandPath('/commandbox/system/config/server-icons/home.png' ) } );
+		serverInfo.trayOptions.prepend( { 'label' : 'Restart Server', 'hotkey':'R', 'action' : "runAsync" , "command" : "box server restart " & "'#serverInfo.name#'", 'image': expandPath('/commandbox/system/config/server-icons/home.png' ), 'workingDirectory': defaultwebroot} );
 
 		serverInfo.trayOptions.prepend( { 'label':'Stop Server', 'action':'stopserver', 'image' : expandPath('/commandbox/system/config/server-icons/stop.png' ) } );
 
 		// Take default options, then append config defaults and server.json trayOptions on top of them (allowing nested overwrite)
 		serverInfo.trayOptions = appendMenuItems( defaults.trayOptions, defaultwebroot, serverInfo.trayOptions );
+		serverInfo.trayOptions = appendMenuItems( serverInfo.trayOptions, defaultwebroot, [] );
```

Comment:
@mgmathus This is a little wonky.  You're adding global config defaults into the server info tray options, then adding the entire thing again to an empty array which means the defaults are technically getting processed twice which is confusing and unnecessary.  Remember, I suggested that the menu items above this portion be put in a temporary variable and then appended in, which should be simple.

```js
var tempOptions = [];
tempOptions.trayOptions.prepend( { 'label':'Open...', 'action':'...' ) } );
tempOptions.trayOptions.prepend( { 'label':'Stop Server', 'action':'stopserver' ) } );
etc...

// Now start appending from scratch
serverInfo.trayOptions = appendMenuItems( tempOptions, defaultwebroot, [] );
serverInfo.trayOptions = appendMenuItems( defaults.trayOptions, defaultwebroot, serverInfo.trayOptions );
serverInfo.trayOptions = appendMenuItems( serverJSON.trayOptions ?: [], defaultServerConfigFileDirectory, serverInfo.trayOptions );
```